pkg/agent/baker.go

```diff
@@ -31,17 +31,15 @@ func InitializeTemplateGenerator() *TemplateGenerator {
 // This function only can be called after the validation of the input NodeBootstrappingConfiguration.
 func (t *TemplateGenerator) getNodeBootstrappingPayload(config *datamodel.NodeBootstrappingConfiguration) string {
 	var customData string
+	// this might seem strange that we're encoding the custom data to a JSON string and then extracting it, but without that serialisation and deserialisation
+	// lots of tests fail.
 	if config.AgentPoolProfile.IsWindows() {
 		customData = getCustomDataFromJSON(t.getWindowsNodeCustomDataJSONObject(config))
+		return base64.StdEncoding.EncodeToString([]byte(customData))
 	} else {
```

Comment:
nit: we can get rid of the else block now that we return in the windows case

Reply:
We can, but that structure of the code implies that windows code paths are less important. Maybe I'll flick the test around and give Linux the early exit instead of Windows :)

Reply:
Done.